src/helpers/unmountRoot.ts

```diff
@@ -0,0 +1,24 @@
+import { reconciler } from '../core/reconciler';
+import { roots } from '../core/roots';
+import { store } from '../store';
+import { type Root } from '../typedefs/Root';
+
+export function unmountRoot(root: Root)
+{
+    store.unmountQueue.delete(root);
+
+    const fiber = root.fiber;
+
+    if (fiber)
+    {
+        reconciler.updateContainer(null, fiber, null, () =>
+        {
+            if (root.applicationState.app)
+            {
+                root.applicationState.app.destroy();
+            }
+
+            roots.delete(root.internalState.canvas!);
```

Comment:
I can say from my testing that `root.internalState.canvas` is undefined here, meaning that `roots` continues to balloon in size. I'll keep looking...

Reply:
I believe `internalState.canvas = canvas;` where the canvas is created on line 78 of `createRoot` should do it!

Reply:
I've got a test that verifies this is the issue. I'm working on the solution now!